chains/btc/listener/util.go

```diff
@@ -0,0 +1,73 @@
+package listener
+
+import (
+	"encoding/hex"
+	"math/big"
+
+	"github.com/ChainSafe/sygma-relayer/e2e/evm"
+	"github.com/btcsuite/btcd/btcjson"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var RESOURCE_ID = evm.SliceTo32Bytes(common.LeftPadBytes([]byte{0}, 31))
+
+const (
+	PubKeyHash       = "pubkeyhash"
+	ScriptHash       = "scripthash"
+	WitnessV0KeyHash = "witness_v0_keyhash"
+	OP_RETURN        = "nulldata"
+)
+
+func DecodeDepositEvent(evt btcjson.TxRawResult, conn Connection, bridge string) (Deposit, bool, error) {
+	amount := big.NewInt(0)
+	isBridgeDeposit := false
+	sender := ""
+	data := ""
+
+	for _, vout := range evt.Vout {
+		// read the OP_RETURN data
+		if vout.ScriptPubKey.Type == OP_RETURN {
+			opReturnData, err := hex.DecodeString(vout.ScriptPubKey.Hex)
+			if err != nil {
+				return Deposit{}, true, err
+			}
+			// Extract OP_RETURN data (excluding OP_RETURN prefix)
+			data = string(opReturnData[2:])
+		}
+
+		if bridge == vout.ScriptPubKey.Address {
+			isBridgeDeposit = true
+
+			if vout.ScriptPubKey.Type == PubKeyHash || vout.ScriptPubKey.Type == ScriptHash || vout.ScriptPubKey.Type == WitnessV0KeyHash {
+				amount.Add(amount, big.NewInt(int64(vout.Value*1e8)))
+			}
+		}
+	}
+
+	if isBridgeDeposit {
```

Comment:
Can you rewrite this as !isBridgeDeposit and return immediately if it isn't?
It is always more clearer to have the optimistic path last.